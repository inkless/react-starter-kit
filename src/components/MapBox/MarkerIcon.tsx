import styled, { StyledProps } from 'styled-components';
import { color, ColorProps } from 'styled-system';
import { ReactComponent as MarkerIcon } from '@src/assets/svg/marker.svg';

export type Props = ColorProps;
type ThemeProps = StyledProps<Props>;

export const StyledMarkerIcon = styled(MarkerIcon)`
  position: absolute;
  left: -14px; /* half of the size */
  top: -35px; /* move the icon to top */
  color: ${(p: ThemeProps) => p.theme.colors.primary.main};
  cursor: pointer;
  ${color}
`;

export default StyledMarkerIcon;
