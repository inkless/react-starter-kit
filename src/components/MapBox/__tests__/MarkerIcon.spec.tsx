import React from 'react';
import { shallow } from 'enzyme';

import MarkerIcon from '../MarkerIcon';

describe('Component MarkerIcon', () => {
  it('should render MarkerIcon correctly', () => {
    const component = shallow(<MarkerIcon />);

    expect(component).toMatchSnapshot();
  });
});
