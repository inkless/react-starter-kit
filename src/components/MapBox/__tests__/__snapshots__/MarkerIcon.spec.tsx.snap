// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component MarkerIcon should render MarkerIcon correctly 1`] = `
<StyledComponent
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-bdVaJa",
        "isStatic": false,
        "rules": Array [
          "
  position: absolute;
  left: -14px; /* half of the size */
  top: -35px; /* move the icon to top */
  color: ",
          [Function],
          ";
  cursor: pointer;
  ",
          [Function],
          "
",
        ],
      },
      "displayName": "Styled(ReactComponent)",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "sc-bdVaJa",
      "target": [Function],
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
/>
`;
